sycl/test-e2e/Graph/event_profiling_info.cpp

```diff
@@ -0,0 +1,112 @@
+
+#include "./graph_common.hpp"
+
+bool verifyProfiling(event Event) {
+  auto Submit =
+      Event.get_profiling_info<sycl::info::event_profiling::command_submit>();
+  auto Start =
+      Event.get_profiling_info<sycl::info::event_profiling::command_start>();
+  auto End =
+      Event.get_profiling_info<sycl::info::event_profiling::command_end>();
+
+  assert((Submit && Start && End) && "Profiling information failed.");
+  assert(Submit < Start);
+  assert(Start < End);
+
+  bool Pass = sycl::info::event_command_status::complete ==
+              Event.get_info<sycl::info::event::command_execution_status>();
+
+  return Pass;
+}
+
+// The test checks that get_profiling_info waits for command asccociated with
+// event to complete execution.
+int main() {
+  device Dev;
+  queue Queue{Dev, sycl::property::queue::enable_profiling()};
+
+  const size_t Size = 10000;
+  int Data[Size] = {0};
+  for (size_t I = 0; I < Size; ++I) {
+    Data[I] = I;
+  }
+  int Values[Size] = {0};
+
+  buffer<int, 1> BufferFrom(Data, range<1>(Size));
+  buffer<int, 1> BufferTo(Values, range<1>(Size));
+
+  buffer<int, 1> BufferA(Data, range<1>(Size));
+  buffer<int, 1> BufferB(Values, range<1>(Size));
+  buffer<int, 1> BufferC(Values, range<1>(Size));
+
+  BufferFrom.set_write_back(false);
+  BufferTo.set_write_back(false);
+  BufferA.set_write_back(false);
+  BufferB.set_write_back(false);
+  BufferC.set_write_back(false);
+  {
+    // buffer copy
+    exp_ext::command_graph CopyGraph{
+        Queue.get_context(),
+        Queue.get_device(),
+        {exp_ext::property::graph::assume_buffer_outlives_graph{},
+         exp_ext::property::graph::assume_data_outlives_buffer{}}};
+    CopyGraph.begin_recording(Queue);
+
+    Queue.submit([&](sycl::handler &Cgh) {
+      accessor<int, 1, access::mode::read, access::target::device> AccessorFrom(
+          BufferFrom, Cgh, range<1>(Size));
+      accessor<int, 1, access::mode::write, access::target::device> AccessorTo(
+          BufferTo, Cgh, range<1>(Size));
+      Cgh.copy(AccessorFrom, AccessorTo);
+    });
+
+    CopyGraph.end_recording(Queue);
+
+    // kernel launch
+    exp_ext::command_graph KernelGraph{
+        Queue.get_context(),
+        Queue.get_device(),
+        {exp_ext::property::graph::assume_buffer_outlives_graph{},
+         exp_ext::property::graph::assume_data_outlives_buffer{}}};
+    KernelGraph.begin_recording(Queue);
+
+    run_kernels(Queue, Size, BufferA, BufferB, BufferC);
+
+    KernelGraph.end_recording(Queue);
+
+    auto CopyGraphExec = CopyGraph.finalize();
+    auto KernelGraphExec = KernelGraph.finalize();
+
+    // Run graphs
+    event CopyEvent = Queue.submit(
+        [&](handler &CGH) { CGH.ext_oneapi_graph(CopyGraphExec); });
+    event KernelEvent = Queue.submit(
+        [&](handler &CGH) { CGH.ext_oneapi_graph(KernelGraphExec); });
+
+    Queue.wait_and_throw();
+
+    // Checks profiling times
+    assert(verifyProfiling(CopyEvent) && verifyProfiling(KernelEvent));
```

Comment:
Can we also test submitting the same graph twice, and verifying that the timestamps are different.

e.g 
```cpp
event KerneEvent1 = Queue.submit(
        [&](handler &CGH) { CGH.ext_oneapi_graph(KernelGraphExec); });
event KerneEvent2 = Queue.submit(
        [&](handler &CGH) { CGH.ext_oneapi_graph(KernelGraphExec); });

// Check that submit/start/end timestamps aren't identical for KernelEvent1 and KernelEvent2

```